### ISSUE TITLE
Bare except: in `fit`

### ISSUE THREAD

**Issue by jbrockmendel** [statsmodels/statsmodels#3779]
https://github.com/statsmodels/statsmodels/blob/master/statsmodels/base/model.py#L439

```
        try:
            hess = lambda params, *args: -self.hessian(params, *args) / nobs
        except:
            hess = None
```

Can we change the `except:` to `except SomethingSpecific:`?  I'm guessing `AttributeError` for models that don't have a `hessian` method.  Are there any other exceptions that can be raised here?  The others that I can think of (e.g. `KeyboardInterrupt`) we definitely _dont_ want to catch.

**Comment by josef-pkt** [comment]
I don't have an overview about when we get warnings and when we get exceptions during optimization.

For the specific case here. #3646 removes the try except completely.
AFAIU, AttributeError would be the only reason that the lambda definition might fail. I assume all likelihoodmodels that use the fit model from base are supposed to provide hessian now.

**Comment by jbrockmendel** [comment]
Fantastic.  Also happy to see the lambdas changed to `def foo`, mainly so that pep8 will stop complaining about it.

As long as attention is being payed to that part of the code, any chance we can clear up this [comment](https://github.com/statsmodels/statsmodels/pull/3646/files#diff-b165b4bd4e10edd0dfb485e47c562b2cR443)

```
    if method == 'newton':
        # TODO: why are score and hess positive?
        def score(params, *args):
            return self.score(params, *args) / nobs
```

**Comment by jbrockmendel** [comment]
Closed by #3646
